2023-09-08 07:37:09 [scrapy] DEBUG: Crawled (200) <GET https://www.google.com/search?q=+apple&hl=en&gl=DK&start=0> (referer: None)
2023-09-08 07:37:09 [scrapy] DEBUG: Crawled (200) <GET https://www.google.com/search?q=+macbook&hl=en&gl=DK&start=0> (referer: None)
2023-09-08 07:37:09 [scrapy] ERROR: Spider error processing <GET https://www.google.com/search?q=+macbook&hl=en&gl=DK&start=0> (referer: None)
Traceback (most recent call last):
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\defer.py", line 257, in iter_errback
    yield next(it)
          ^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\offsite.py", line 28, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\referer.py", line 353, in <genexpr>
    return (self._set_referer(r, response) for r in result or ())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\urllength.py", line 27, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\depth.py", line 31, in <genexpr>
    return (r for r in result or () if self._filter(r, response, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "D:\Outputs\Django Outputs\scraper\spiders\plaspider.py", line 91, in parse
    item['user_id'] = self.user.id
    ^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\item.py", line 85, in __setitem__
    raise KeyError(f"{self.__class__.__name__} does not support field: {key}")
KeyError: 'ScraperItem does not support field: user'
2023-09-08 07:37:09 [scrapy] DEBUG: Crawled (200) <GET https://www.google.com/search?q=+ps5&hl=en&gl=DK&start=0> (referer: None)
2023-09-08 07:37:09 [scrapy] ERROR: Spider error processing <GET https://www.google.com/search?q=+ps5&hl=en&gl=DK&start=0> (referer: None)
Traceback (most recent call last):
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\defer.py", line 257, in iter_errback
    yield next(it)
          ^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\offsite.py", line 28, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\referer.py", line 353, in <genexpr>
    return (self._set_referer(r, response) for r in result or ())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\urllength.py", line 27, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\depth.py", line 31, in <genexpr>
    return (r for r in result or () if self._filter(r, response, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "D:\Outputs\Django Outputs\scraper\spiders\plaspider.py", line 91, in parse
    item['user_id'] = self.user.id
    ^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\item.py", line 85, in __setitem__
    raise KeyError(f"{self.__class__.__name__} does not support field: {key}")
KeyError: 'ScraperItem does not support field: user'
2023-09-08 07:37:09 [scrapy] DEBUG: Crawled (200) <GET https://www.google.com/search?q=+iphone+13&hl=en&gl=DK&start=0> (referer: None)
2023-09-08 07:37:09 [scrapy] ERROR: Spider error processing <GET https://www.google.com/search?q=+iphone+13&hl=en&gl=DK&start=0> (referer: None)
Traceback (most recent call last):
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\defer.py", line 257, in iter_errback
    yield next(it)
          ^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\offsite.py", line 28, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\referer.py", line 353, in <genexpr>
    return (self._set_referer(r, response) for r in result or ())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\urllength.py", line 27, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\depth.py", line 31, in <genexpr>
    return (r for r in result or () if self._filter(r, response, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "D:\Outputs\Django Outputs\scraper\spiders\plaspider.py", line 91, in parse
    item['user_id'] = self.user.id
    ^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\item.py", line 85, in __setitem__
    raise KeyError(f"{self.__class__.__name__} does not support field: {key}")
KeyError: 'ScraperItem does not support field: user'
2023-09-08 07:37:10 [scrapy] DEBUG: Crawled (200) <GET https://www.google.com/search?q=iphone+14&hl=en&gl=DK&start=0> (referer: None)
2023-09-08 07:37:10 [scrapy] ERROR: Spider error processing <GET https://www.google.com/search?q=iphone+14&hl=en&gl=DK&start=0> (referer: None)
Traceback (most recent call last):
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\defer.py", line 257, in iter_errback
    yield next(it)
          ^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\utils\python.py", line 312, in __next__
    return next(self.data)
           ^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\offsite.py", line 28, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\referer.py", line 353, in <genexpr>
    return (self._set_referer(r, response) for r in result or ())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\urllength.py", line 27, in <genexpr>
    return (r for r in result or () if self._filter(r, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\spidermiddlewares\depth.py", line 31, in <genexpr>
    return (r for r in result or () if self._filter(r, response, spider))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\core\spidermw.py", line 104, in process_sync
    for r in iterable:
  File "D:\Outputs\Django Outputs\scraper\spiders\plaspider.py", line 91, in parse
    item['user_id'] = self.user.id
    ^^^^^^^^^^^^
  File "C:\Users\Muhammad Umer\AppData\Local\Programs\Python\Python311\Lib\site-packages\scrapy\item.py", line 85, in __setitem__
    raise KeyError(f"{self.__class__.__name__} does not support field: {key}")
KeyError: 'ScraperItem does not support field: user'
2023-09-08 07:37:10 [scrapy] INFO: Closing spider (finished)
2023-09-08 07:37:10 [scrapy] INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 2102,
 'downloader/request_count': 5,
 'downloader/request_method_count/GET': 5,
 'downloader/response_bytes': 1241641,
 'downloader/response_count': 5,
 'downloader/response_status_count/200': 5,
 'elapsed_time_seconds': 22.867019,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2023, 9, 8, 2, 37, 10, 741153),
 'httpcompression/response_bytes': 4730586,
 'httpcompression/response_count': 5,
 'log_count/DEBUG': 5,
 'log_count/ERROR': 4,
 'log_count/INFO': 8,
 'response_received_count': 5,
 'scheduler/dequeued': 5,
 'scheduler/dequeued/memory': 5,
 'scheduler/enqueued': 5,
 'scheduler/enqueued/memory': 5,
 'spider_exceptions/KeyError': 4,
 'start_time': datetime.datetime(2023, 9, 8, 2, 36, 47, 874134)}
2023-09-08 07:37:10 [scrapy] INFO: Spider closed (finished)
